src/components/modals/ContactListModal.tsx

```diff
@@ -55,6 +57,10 @@ export function ContactListModal({ bridge, contactType, contactName }: Props) {
 
   const handleSubmitEditing = (contactName: string) => bridge.resolve({ contactName, thumbnailPath: null })
 
+  React.useEffect(() => {
```

Comment:
It looks like this will not make any attempt to grab all the contacts before the modal is shown. While the permission might be approved, nothing will refresh the `ContactsLoader` and force a refresh of contacts. However this issue seems to be even outside of this commit. However this commit may exacerbate the issue

Reply:
On fresh run of the app, before any changes to permissions settings has been
made, the app does properly refresh the contacts and thumbnails.